src/core/utilities/mpi_utils.hpp

```diff
@@ -57,6 +61,46 @@ auto mpi_type_for()
     // don't return anything = compile failure if tried to use this function
 }
 
+
+template<typename Fn, typename... Args>
+auto get_from_rank_0_only(Fn&& fn, Args&&... args)
```

Comment:
`all_get_from(Fn&& fn, Args&&... args, int rank = 0)` ?

Reply:
I don't think you can do that, maybe 

```
all_get_from(int rank, Fn&& fn, Args&&... args) {}

all_get_from_rank_0(Fn&& fn, Args&&... args) {
 return all_get_from(0, ...)
}
```

Reply:
yeah ok, as you wish, the rank think is not essential. I prefer the `all_get_from....` however, it conveys better the collective aspect of the com.

Reply:
to me, that it's in the MPI namespace, kinda implies it's collective, otherwise why would it be via "mpi"
but anyway, it is now `all_get_from`